content/docs/pipelines/sdk/python-based-visualizations.md

```diff
@@ -0,0 +1,177 @@
+            # Pipelines UI and append DataFrame to an array of DataFrames
+            dfs = []
+            for f in file_io.get_matching_files(source):
+                dfs.append(pd.read_csv(f))
+            ...
+            # Get a path from the Pipelines UI and create a DataFrame
+            df = pd.read_csv(source)
+            ...
+            ```
+            * Additional details about how this is implemented can be found
+            [here](https://github.com/kubeflow/pipelines/blob/master/backend/
+            src/apiserver/visualization/server.py#L127).
+        * To obtain additional variables from the Pipelines UI, you can use
+        the following syntax:
+
+            ```python
+            ...
+            # Check if a variable is provided
+            # You should always check if the desired variable is provided before
+            # accessing it because accessing a key from a dict when it does not
+            # exist will result in an exception being raised.
+            if "key" in variables:
+              # Variable of name key is provided.
+              # Use the value of the specified variable to manipulate the way
+              # a visualization is generated here.
+              pass
+            else:
+              # Variable of name key is not provided.
+              # You can provide a default operation here if a variable is not
+              # provided or throw an error if the variable must be provided.
+              pass
+            ...
+            # Accessing a variable
+            key = variables["key"]
+            # or
+            key = variables.get("key")
+            ...
+            ```
+            * Additional details about how this is implemented can be found
+            [here](https://github.com/kubeflow/pipelines/blob/master/backend/
+            src/apiserver/visualization/exporter.py#L93).
+9. Add any new dependencies to the **requirements.txt** file in the
+`backend/src/apiserver/visualization` directory.
+    * This only needs to be updated if new dependencies are added.
+    * If new dependencies are added, the
+    [**third_party_licenses.csv**](https://github.com/kubeflow/pipelines/blob/
+    master/backend/src/apiserver/visualization/third_party_licenses.csv) file
+    must also be updated.
+        
+        ```csv
+        package_name,url_to_package_license,license_name
+        ```
+        * `package_name`: the name of the package on [pypi](https://pypi.org/)
+        * `url_to_package_license`: the url where the license of the package can
+        be downloaded from
+        * `license_name`: name of package license
+        * Examples for all the columns can be found [here](https://github.com/
+        kubeflow/pipelines/blob/master/backend/src/apiserver/visualization/
+        third_party_licenses.csv).
+10. Submit these changes as a PR or build docker image for usage within your
+cluster.
+
+## Using Predefined Visualizations
```

Comment:
A screenshot or two would be very helpful, here and in the next section.

Reply:
How detailed should these screenshots be? Should there be one for each step and if so, should there be arrows or an overlay to indicate what portion of the screenshot is being referenced?